### PR TITLE
Align thread pool info to thread pool configuration

### LINKED ISSUE #29113: No core value reported for scaling thread pool type
**Elasticsearch version** (`bin/elasticsearch --version`): >= 5.x

**Description of the problem including expected versus actual behavior**:

This is probably documentation related issue. There are a few places where the documentation mentions that the scaling thread pool uses the `core` value for the minimum value. For example:

- [Doc for scaling thread pool](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/modules-threadpool.html#_literal_scaling_literal): "... between the value of the `core` and `max` parameters"
- [5.x breaking changes for thread pool settings](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/breaking_50_settings_changes.html#_threadpool_settings): "... has been changed from `min` to `core`"

However, in the output I see only `min` value where the `core` should be expected:

```json
// http://10.8.245.12:9200/_nodes/_local/thread_pool

{
  "_nodes": {
    "total": 1,
    "successful": 1,
    "failed": 0
  },
  "cluster_name": "xxxxxxxx",
  "nodes": {
    "DK0UQ_fFTZ6fz0H2v9XsKw": {
      "name": "xxxxxxxx-1",
      "transport_address": "172.16.220.11:9300",
      "host": "172.16.220.11",
      "ip": "172.16.220.11",
      "version": "5.5.2",
      "build_hash": "b2f0c09",
      "roles": [
        "master",
        "data",
        "ingest"
      ],
      "thread_pool": {
        "force_merge": {
          "type": "fixed",
          "min": 1,
          "max": 1,
          "queue_size": -1
        },
        "fetch_shard_started": {
          "type": "scaling",
          "min": 1,
          "max": 8,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "listener": {
          "type": "fixed",
          "min": 2,
          "max": 2,
          "queue_size": -1
        },
        "index": {
          "type": "fixed",
          "min": 4,
          "max": 4,
          "queue_size": 200
        },
        "refresh": {
          "type": "scaling",
          "min": 1,
          "max": 2,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "generic": {
          "type": "scaling",
          "min": 4,
          "max": 128,
          "keep_alive": "30s",
          "queue_size": -1
        },
        "warmer": {
          "type": "scaling",
          "min": 1,
          "max": 2,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "search": {
          "type": "fixed",
          "min": 7,
          "max": 7,
          "queue_size": 1000
        },
        "flush": {
          "type": "scaling",
          "min": 1,
          "max": 2,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "fetch_shard_store": {
          "type": "scaling",
          "min": 1,
          "max": 8,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "management": {
          "type": "scaling",
          "min": 1,
          "max": 5,
          "keep_alive": "5m",
          "queue_size": -1
        },
        "get": {
          "type": "fixed",
          "min": 4,
          "max": 4,
          "queue_size": 1000
        },
        "bulk": {
          "type": "fixed",
          "min": 4,
          "max": 4,
          "queue_size": 200
        },
        "snapshot": {
          "type": "scaling",
          "min": 1,
          "max": 2,
          "keep_alive": "5m",
          "queue_size": -1
}}}}}
```

### PR DESCRIPTION
**Pull request by jasontedor**
Today we report thread pool info using a common object. This means that we use a shared set of terminology that is not consistent with the terminology used to the configure thread pools. This holds in particular for the minimum and maximum number of threads in the thread pool where we use the following terminology:
```
 thread pool info | fixed | scaling
 min                core    size
 max                max     size
```

This commit changes the display of thread pool info to be dependent on the type of the thread pool so that we can align the terminology in the output of thread pool info with the terminology used to configure a thread pool.

Closes #29113

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java [hunk 1]
```diff
@@ -239,16 +251,7 @@ public Counter estimatedTimeInMillisCounter() {
...
-        List<Info> infos = new ArrayList<>();
-        for (ExecutorHolder holder : executors.values()) {
-            String name = holder.info.getName();
-            // no need to have info on "same" thread pool
-            if ("same".equals(name)) {
-                continue;
-            }
-            infos.add(holder.info);
-        }
-        return new ThreadPoolInfo(infos);
```
Comment (jasontedor) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java]: There is no need to recreate this info (garbage!) on every invocation, it is immutable.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by lukas-vlcek** [comment]
Thanks for this.
Isn't this kind of a breaking change? People parsing JSON stats will need to update on their end. Should be mentioned in release notes.

**Comment by jasontedor** [comment]
That is a fair point, thanks for raising it. I will add something to the migration notes and remove this change from 6.3.0.

**Comment by spinscale** [comment]
the `_cat` API has a `thread_pool` endpoint as well, which IIRC uses `min`, `max`, and `size` as well

**Pull request merged**